graylog2-web-interface/src/views/logic/aggregationbuilder/visualizations/HeatmapVisualizationConfig.ts

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as Immutable from 'immutable';
+
+import VisualizationConfig from 'views/logic/aggregationbuilder/visualizations/VisualizationConfig';
+
+type ColorScale =
+    'Greys'
+  | 'YlGnBu'
+  | 'Greens'
+  | 'YlOrRd'
+  | 'Bluered'
+  | 'RdBu'
+  | 'Reds'
+  | 'Blues'
+  | 'Picnic'
+  | 'Rainbow'
+  | 'Portland'
+  | 'Jet'
+  | 'Hot'
+  | 'Blackbody'
+  | 'Earth'
+  | 'Electric'
+  | 'Viridis'
+  | 'Cividis';
+
+export const COLORSCALES = ['Greys', 'YlGnBu', 'Greens', 'YlOrRd', 'Bluered', 'RdBu', 'Reds', 'Blues', 'Picnic', 'Rainbow', 'Portland', 'Jet', 'Hot', 'Blackbody', 'Earth', 'Electric', 'Viridis', 'Cividis'];
```

Comment:
To avoid duplicating all values in the type, we can do:
```suggestion
export const COLORSCALES = ['Greys', 'YlGnBu', 'Greens', 'YlOrRd', 'Bluered', 'RdBu', 'Reds', 'Blues', 'Picnic', 'Rainbow', 'Portland', 'Jet', 'Hot', 'Blackbody', 'Earth', 'Electric', 'Viridis', 'Cividis'];

type ColorScale = typeof COLORSCALES[number];
```